app/com/linkedin/drelephant/tuning/engine/SparkHBTParamRecommender.java

```diff
@@ -360,4 +430,62 @@ private int getMemoryIncreaseForGC() {
       return 0;
     }
   }
+
+  /**
+   * This method suggest three parameters : spark.dynamicAllocation.minExecutors, spark.dynamicAllocation.maxExecutors
+   *  and spark.executor.instances
+   * For now we are changing these parameters only in case we are changing number of executor core
+   */
+  private void suggestNumExecutorParameters() {
+    if (suggestedCore != lastRunExecutorCore) {
+      if (lastRunDynamicAllocationMinExecutors != null) {
+        suggestedDynamicAllocationMinExecutors =
+            (int) Math.ceil(lastRunDynamicAllocationMinExecutors * lastRunExecutorCore / suggestedCore);
+      }
+      if (lastRunDynamicAllocationMaxExecutors != null) {
+        suggestedDynamicAllocationMaxExecutors =
+            (int) Math.ceil(lastRunDynamicAllocationMaxExecutors * lastRunExecutorCore / suggestedCore);
+      }
+      if (lastRunExecutorInstances != null) {
+        suggestedExecutorInstances = (int) Math.ceil(lastRunExecutorInstances * lastRunExecutorCore / suggestedCore);
+      }
+      validateNumExecutorParameters();
+    } else {
+      suggestedDynamicAllocationMinExecutors = lastRunDynamicAllocationMinExecutors;
+      suggestedDynamicAllocationMaxExecutors = lastRunDynamicAllocationMaxExecutors;
+      suggestedExecutorInstances = lastRunExecutorInstances;
+    }
+  }
+
+  /**
+   * Validating executor parameters
+   * Anything should not be less than 1 and more than MAX_EXECUTORS
+   */
+  private void validateNumExecutorParameters() {
+    if (suggestedDynamicAllocationMinExecutors != null) {
```

Comment:
The code below seems very similar i.e. has similar checks. Can this be factored out to a common method which can be reused

Reply:
Now there is only one validation.